src/adapter/src/coord/ddl.rs

```diff
@@ -747,29 +753,40 @@ impl Coordinator {
             })),
             SINCE_GRANULARITY,
         );
+        self.update_retention(policy, |item| item.is_retained_metrics_object())
+    }
+
+    fn update_default_retention(&mut self) {
+        let lag = self.catalog().system_config().default_retention_timestamp();
+        let policy = ReadPolicy::lag_writes_by(lag, SINCE_GRANULARITY);
+        self.update_retention(policy, |item| item.uses_default_retention())
+    }
+
+    fn update_retention(
+        &mut self,
+        policy: ReadPolicy<Timestamp>,
+        update_item: impl Fn(&CatalogItem) -> bool,
+    ) {
         let storage_policies = self
-            .catalog()
-            .entries()
-            .filter(|entry| {
-                entry.item().is_retained_metrics_object()
-                    && entry.item().is_compute_object_on_cluster().is_none()
+            .storage_read_capabilities
+            .iter()
+            .filter_map(|(id, _cap)| {
+                update_item(self.catalog().get_entry(id).item()).then(|| (*id, policy.clone()))
             })
-            .map(|entry| (entry.id(), policy.clone()))
             .collect::<Vec<_>>();
+
         let compute_policies = self
-            .catalog()
-            .entries()
-            .filter_map(|entry| {
-                if let (true, Some(cluster_id)) = (
-                    entry.item().is_retained_metrics_object(),
-                    entry.item().is_compute_object_on_cluster(),
-                ) {
-                    Some((cluster_id, entry.id(), policy.clone()))
-                } else {
-                    None
-                }
+            .compute_read_capabilities
+            .iter()
+            .filter(|(id, _)| !id.is_transient())
+            .filter_map(|(id, _cap)| {
```

Comment:
If we're removing all of the transient IDs, then why do we need to further filter it?

Reply:
i need to check cluster_id is present and update_item returns true. Are you thinking it'd be better to have just one filter call that checks those things and !id.is_transient()?

Reply:
Oh, I see, ok that looks fine then.